test/unit/spec/common/ui-Spec.js

```diff
@@ -579,4 +579,201 @@ define(['common/ui'], (UI) => {
+                        okLabel: bs,
+                        bsClass: bs,
+                        doThisFirst: () => {
+                            expect(document.querySelector('.modal-header')).toHaveClass(`bg-${bs}`);
+                            expect(document.querySelector('.modal-title')).toHaveClass(
+                                `text-${bs}`
+                            );
+                            buttonCheck(bs);
+                        },
+                    });
+                });
+            });
+
+            it('can use the bootstrap primary theme', async () => {
+                const bs = 'primary';
+                await UI.showConfirmDialog({
+                    body: `${bs} test dialog`,
+                    title: bs,
+                    okLabel: bs,
+                    bsClass: bs,
+                    doThisFirst: () => {
+                        expect(document.querySelector('.modal-header')).toHaveClass(`bg-${bs}`);
+                        // uses the text colour from the bg
+                        expect(document.querySelector('.modal-title')).not.toHaveClass(
+                            `text-${bs}`
+                        );
+                        buttonCheck(bs);
+                    },
+                });
+            });
+
+            it('can be unthemed', async () => {
+                const bs = 'primary';
+                await UI.showConfirmDialog({
+                    body: `unthemed test dialog`,
+                    title: 'No theme',
+                    doThisFirst: () => {
+                        // no background!
+                        expect(document.querySelector('.modal-header')).not.toHaveClass(`bg-${bs}`);
+                        expect(document.querySelector('.modal-title')).toHaveClass(`text-${bs}`);
+                        buttonCheck(bs);
+                    },
+                });
+            });
+        });
+
+        describe('showConfirmDialog', () => {
+            const clickLocation = {
+                '.modal-header .close': false,
+                '[data-element="ok"]': true,
+                '[data-element="cancel"]': false,
+                // modal backdrop
+                '.modal': false,
+            };
+            const keyCode = {
+                13: true,
+                27: false,
+            };
+            Object.keys(clickLocation).forEach((loc) => {
+                it(`returns ${clickLocation[loc]} when clicking on ${loc}`, async () => {
+                    await UI.showConfirmDialog({
+                        title: 'showConfirmDialog',
+                        body: 'blah blah blah',
+                        doThisFirst: () => {
+                            document.querySelector(loc).click();
+                        },
+                    }).then((res) => {
+                        expect(res).toBe(clickLocation[loc]);
+                        expect(document.querySelector('.modal-dialog')).toBeNull();
+                    });
```

Comment:
testing these modals was a PITA because they return a Promise that resolves when the modal is dismissed, but there was no way to actually dismiss them -- hence having to inject the modal dismissal using the `doThisFirst` argument.